core/src/main/java/hudson/tasks/BatchFile.java

```diff
@@ -71,9 +94,28 @@ public String getDisplayName() {
             return Messages.BatchFile_DisplayName();
         }
 
-        @Override
-        public Builder newInstance(StaplerRequest req, JSONObject data) {
-            return new BatchFile(data.getString("command"));
+        /**
+         * Performs on-the-fly validation of the errorlevel.
+         */
+        @Restricted(DoNotUse.class)
+        public FormValidation doCheckUnstableReturn(@QueryParameter String value) {
+            value = Util.fixEmptyAndTrim(value);
+            if (value == null) {
+                return FormValidation.ok();
+            }
+            long unstableReturn;
+            try {
+                unstableReturn = Long.parseLong(value);
+            } catch (NumberFormatException e) {
+                return FormValidation.error(hudson.model.Messages.Hudson_NotANumber());
+            }
+            if (unstableReturn == 0) {
+                return FormValidation.warning(hudson.tasks.Messages.BatchFile_invalid_exit_code_zero());
+            }
+            if (unstableReturn < Integer.MIN_VALUE || unstableReturn > Integer.MAX_VALUE) {
+                return FormValidation.error(hudson.tasks.Messages.BatchFile_invalid_exit_code_range(unstableReturn));
```

Comment:
I tested a simple batch file with a single line: 'exit /b -2147483648' and the errorlevel is set to '-2147483648'. It works on win7, etc. I think it's better not to limit the available range, although hardly used.